spec/classes/snmp_init_spec.rb

```diff
@@ -630,6 +630,13 @@
           it { is_expected.not_to contain_file('snmptrapd.sysconfig') }
           it { is_expected.not_to contain_service('snmptrapd') }
         end
+        case facts[:os]['release']['major']
+        when '9', '10', '16.04', '18.04', '20.04'
+          let(:facts) { facts.merge(:service_provider => 'systemd') }
```

Comment:
This is not a good place for `let(:facts)`. It should really be done right below the `context` since `let` blocks define a context. You can also write it as:
```suggestion
          let(:facts) { facts.merge(service_provider: 'systemd') }
```

Reply:
If we drop it under the context the specs are not exactly true then? as older operating systems will pass as systemd in tests?